File: csharp/main/LiteralConstant.cs
using System;

namespace Deltix.Luminary
{
	public class LiteralConstant : Literal, ILiteral<ConstantDef>
	{
		public LiteralConstant(ConstantDef value)
			: base(LiteralKind.Constant)
		{
			Value = value;
		}

		public ConstantDef Value { get; }

		public override String ToString()
		{
			return $"{Value.Owner.Name}.{Value.Name}";
		}
	}
}

File: java/src/main/java/deltix/luminary/LiteralConstant.java
package deltix.luminary;

public class LiteralConstant extends Literal {
    private final ConstantDef value;

    public LiteralConstant(ConstantDef value) {
        super(LiteralKind.CONSTANT);
        this.value = value;
    }

    public ConstantDef getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value.getOwner().getName() + "." + value.getName();
    }
}
